x-pack/legacy/plugins/alerting/server/alerts_client.ts

```diff
@@ -436,6 +452,8 @@ export class AlertsClient {
     return {
       id,
       ...rawAlert,
+      updatedAt: updatedAt ? new Date(updatedAt) : null,
+      createdAt: new Date(rawAlert.createdAt!),
```

Comment:
Side note regarding the `!`, though not related to this PR: When I fixed the AAD issues in the `update` function (https://github.com/elastic/kibana/pull/43712/files#diff-98aaa0f889f7ee8284542a3d80348c4eR213) this `getAlertFromRaw` function should now receive a full `rawAlert` object all the time without the need of `Partial<...>`.

Reply:
Not necessarily... as we still need a Partial for updates... SavedObject's API is tricky.